prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -2312,26 +2312,28 @@ void rxr_pkt_handle_write_rta_send_completion(struct rxr_ep *ep, struct rxr_pkt_
 	rxr_cq_handle_send_completion(ep, tx_entry);
 }
 
-static int rxr_write_atomic_hmem(struct efa_mr *efa_mr, struct iovec *iov, char* data,
+static int rxr_write_atomic_hmem(struct efa_mr *efa_mr, struct iovec *dst, char *data,
                                  size_t dtsize, int op, int dt)
 {
-	char host_data[iov->iov_len];
+	char *host_data = (char *) malloc(dst->iov_len);
 	uint64_t device = efa_mr->peer.device.reserved;
 	int err;
 
 	/* Step 1: Copy data from device to temporary host buffer */
-	err = ofi_copy_from_hmem(efa_mr->peer.iface, device, host_data, iov->iov_base, iov->iov_len);
+	err = ofi_copy_from_hmem(efa_mr->peer.iface, device, host_data, dst->iov_base, dst->iov_len);
 	if (OFI_UNLIKELY(err)) {
+		free(host_data);
 		return err;
 	}
```

Comment:
(1/2) Might be easier to maintain:

```suggestion
	if (OFI_UNLIKELY(err)) {
		goto done;
	}
```

Reply:
I agree with this. Though I will use `exit` instead of `done`. which